source/common/network/connection_impl.cc

```diff
@@ -438,6 +445,27 @@ void ConnectionImpl::onWriteReady() {
     ASSERT(0 == rc);
     UNREFERENCED_PARAMETER(rc);
 
+#ifdef __APPLE__
+    // When a connect() is performed on a non-blocking socket, it'll return -1/EINPROGRESS
```

Comment:
This PR LGTM with this exception as well. Just some better citations would be sufficient if there is no way to improve on this.